packages/circuit-ui/components/ImageInput/ImageInput.mdx

```diff
@@ -17,8 +17,34 @@ The ImageInput component allows users to upload images. It can be used on its ow
 
 Refer to the `Stateful` component story for a simplified code usage example.
 
-- **Do** use the `onChange` callback to immediately trigger an image upload when the user has selected one, even when the component is being used as part of a form. Faster form submissions add to a delightful user experience.
-- **Do not** use the ImageInput component if the image is read-only for the user. Instead, use the visual component from the ImageInput's `component` prop by itself, for example the [Avatar](Components/Avatar).
+### Uploading an image
+
+When an ImageInput is used as part of a form, there are two possible patterns for uploading the image picked by a user:
+
+1. upload the image as soon as the user selected it (`onChange`)
+2. upload the image when the form is submitted (`onSubmit`)
+
+If possible, prefer the first approach: APIs tend to expect image URLs, not blobs, so uploading an image during form submission is likely to require request chaining (a first request uploads the image to an object storage provider and returns an object URL, a second request sends the object URL along with other form data to some backend service). (If the backend service you're working with handles object storage, choose the second approach.)
+
+### Accepting HEIF image uploads
```

Comment:
This while section is adapted from https://github.com/sumup-oss/circuit-ui/issues/1773#issuecomment-1285601910